django_webapps/scripts/db_population/parsers/common_parsers.py

```diff
@@ -58,13 +61,26 @@ def parse_timezone(tz_str: str) -> datetime.tzinfo | None:
     return None
 
 
+def find_any_email(email_str: str) -> str | None:
+    regex = re.compile(
+        r"([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\"([]!#-[^-~ \t]|(\\[\t -~]))+\")@([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\[[\t -Z^-~]*])"  # noqa: E501
```

Comment:
Django's `EmailField` could have different requirements. In the "dry" mode, the way it is now, we will not catch this error. I think we could [modify dry mode](https://adamj.eu/tech/2022/10/13/dry-run-mode-for-data-imports-in-django/) so that it involves Django models even if `dry` is `True`. This way we could avoid some other pitfalls we don't yet know about.

Otherwise our dry run will succeed, but then the actual writing to database might fail.

Reply:
Yes, you are right, will do a refactoring, but here it is enough to just use validate_email for every string since django uses the same lib